rust/hyperlane-core/src/types/message.rs

```diff
@@ -34,6 +36,34 @@ pub struct HyperlaneMessage {
     pub body: Vec<u8>,
 }
 
+impl Debug for HyperlaneMessage {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "HyperlaneMessage {{ id: {:?}, version: {}, nonce: {}, origin: {}, sender: {}, destination: {}, recipient: {}, body: 0x{} }}",
+            self.id(),
+            self.version,
+            self.nonce,
+            self.origin,
+            fmt_address_for_domain(self.origin, self.sender),
+            self.destination,
+            fmt_address_for_domain(self.destination, self.recipient),
+            hex::encode(&self.body)
+        )
+    }
+}
+
+impl Display for HyperlaneMessage {
```

Comment:
Where is this used? Wondering if origin/destination should be included as well (and maybe recipient and sender)

Reply:
Used in some logs with the `%` to reduce redundancy. Not all logs need the full message for context.

Reply:
> Not all logs need the full message for context.

Totally agree, especially not the body. In my experience the destination/recipient can be useful for debugging (did I send to the wrong address? what is their ISM?), but happy to leave as is and if we find it useful later we can always add